pkg/backend/apply.go

```diff
@@ -130,14 +130,35 @@ func PreviewThenPrompt(ctx context.Context, kind apitype.UpdateKind, stack Stack
 		return plan, changes, nil
 	}
 
+	stats := computeUpdateStats(events)
+
 	infoPrefix := "\b" + op.Opts.Display.Color.Colorize(colors.SpecWarning+"info: "+colors.Reset)
 	if kind != apitype.UpdateUpdate {
 		// If not an update, we can skip displaying warnings
-	} else if countResources(events) == 0 {
+	} else if stats.numNonStackResources == 0 {
 		// This is an update and there are no resources being CREATED
 		fmt.Print(infoPrefix, "There are no resources in your stack (other than the stack resource).\n\n")
 	}
 
+	// Warn user if an update is going to leave untracked resources in the environment.
+	switch kind {
+	case apitype.UpdateUpdate, apitype.PreviewUpdate, apitype.DestroyUpdate:
+		if len(stats.retainedResources) == 0 {
+			// No resources. Skip.
+			break
+		}
+		fmt.Printf(
+			infoPrefix+"This update will leave %d resource(s) untracked in your environment:\n",
+			len(stats.retainedResources))
```

Comment:
If we're already using a printf here, we don't need to append infoPrefix into the string. We can `%s`.
```suggestion
			"%sThis update will leave %d resource(s) untracked in your environment:\n",
			infoPrefix, len(stats.retainedResources))
```